i18n/scripts/extract-keys.ts

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable  */
+// disable-translation
+import fs from 'fs'
+import glob from 'glob'
+import path from 'path'
+import { str as crc32 } from 'crc-32'
+import { TranslateAbleBasePath, TranslateAbleGlobPatters } from '../constants'
+import { KeyValueRefsType } from '../types'
+import { program } from "commander";
+import translated_keys from "../../src/translations/ach.json"
+
+program
+    .version('0.1.0')
+    .description('Build translation source.')
+    .option('-v, --verbose', 'Displays the list of paths to be compiled')
+    .parse(process.argv)
+
+
+const getHashedValue = (value: string) => `${crc32(value)}`
+const action = process.argv[2]
+
+
+const regex = new RegExp(/(_t_)(?<pure_text>.*?)(_t_)/g)
+const foundFilePaths: string[] = []
+const messages: KeyValueRefsType = {}
+const derivedKeys = {}
+let tempHashKey = ""
+
+function extractTranslations() {
+    // Find all file types listed in `globs`
+    try {
+        for (let i = 0; i < TranslateAbleGlobPatters.length; i++) {
+            let filesFound = glob.sync(`${TranslateAbleBasePath}${TranslateAbleGlobPatters[i]}`)
+            filesFound = filesFound.filter((path) => path.indexOf('__tests__') === -1)
+            foundFilePaths.push(...filesFound)
+        }
+        for (let i = 0; i < foundFilePaths.length; i++) {
+            const file = fs.readFileSync(foundFilePaths[i], 'utf8')
+            if (!file.includes("disable-translation")) {
+                let result = regex.exec(file)
+                while (result != null) {
+                    const pureText = result.groups.pure_text
+                    // remove all special characters
+                    let key = pureText.replace(/[`~!@#$^&*()_|+\-=?;:'",‘’.<>\{\}\[\]\\\/””–]/g, '')
```

Comment:
Keep all the regex text in a descriptive variable name , so it will be easy to update/modify in the future